packages/browser/src/tracing/linkedTraces.ts

```diff
@@ -0,0 +1,239 @@
+import type { Client, PropagationContext, Span } from '@sentry/core';
+import {
+  type SpanContextData,
+  getCurrentScope,
+  getRootSpan,
+  logger,
+  SEMANTIC_ATTRIBUTE_SENTRY_PREVIOUS_TRACE_SAMPLE_RATE,
+  SEMANTIC_ATTRIBUTE_SENTRY_SAMPLE_RATE,
+  SEMANTIC_LINK_ATTRIBUTE_LINK_TYPE,
+  spanToJSON,
+} from '@sentry/core';
+import { DEBUG_BUILD } from '../debug-build';
+import { WINDOW } from '../exports';
+
+export interface PreviousTraceInfo {
+  /**
+   * Span context of the previous trace's local root span
+   */
+  spanContext: SpanContextData;
+
+  /**
+   * Timestamp in seconds when the previous trace was started
+   */
+  startTimestamp: number;
+
+  /**
+   * sample rate of the previous trace
+   */
+  sampleRate: number;
+
+  /**
+   * The sample rand of the previous trace
+   */
+  sampleRand: number;
+}
+
+// 1h in seconds
+export const PREVIOUS_TRACE_MAX_DURATION = 3600;
+
+// session storage key
+export const PREVIOUS_TRACE_KEY = 'sentry_previous_trace';
+
+export const PREVIOUS_TRACE_TMP_SPAN_ATTRIBUTE = 'sentry.previous_trace';
+
+/**
+ * Takes care of linking traces and applying the (consistent) sampling behavoiour based on the passed options
+ * @param options - options for linking traces and consistent trace sampling (@see BrowserTracingOptions)
+ * @param client - Sentry client
+ */
+export function linkTraces(
+  {
+    linkPreviousTrace,
+    consistentTraceSampling,
+  }: {
+    linkPreviousTrace: 'session-storage' | 'in-memory';
+    consistentTraceSampling: boolean;
+  },
+  client: Client,
+): void {
+  const useSessionStorage = linkPreviousTrace === 'session-storage';
+
+  let inMemoryPreviousTraceInfo = useSessionStorage ? getPreviousTraceFromSessionStorage() : undefined;
+
+  client.on('spanStart', span => {
+    if (getRootSpan(span) !== span) {
+      return;
+    }
+
+    const oldPropagationContext = getCurrentScope().getPropagationContext();
+    inMemoryPreviousTraceInfo = addPreviousTraceSpanLink(inMemoryPreviousTraceInfo, span, oldPropagationContext);
+
+    if (useSessionStorage) {
+      storePreviousTraceInSessionStorage(inMemoryPreviousTraceInfo);
+    }
+  });
+
+  let isFirstTraceOnPageload = true;
+  if (consistentTraceSampling) {
+    /*
+    When users opt into `consistentTraceSampling`, we need to ensure that we propagate
+    the previous trace's sample rate and rand to the current trace. This is necessary because otherwise, span
+    metric extrapolation is inaccurate, as we'd propagate a too high sample rate for the subsequent traces.
```

Comment:
```suggestion
    metric extrapolation is inaccurate, as we'd propagate too high a sample rate for the subsequent traces.
```

or "a sampling rate that is too high"

Reply:
I asked Jesse (native speaker) and he proposed this: we’d propagate too high of a sample rate